dLux/apertures.py

```diff
@@ -0,0 +1,1430 @@
+        """
+        Returns
+        -------
+        max_radius : float, meters
+            The distance from the centre of the hexagon to one of 
+            the vertices.
+        """
+        return self.rmax
+
+
+    def _aperture(self : Layer) -> Array:
+        """
+        Generates an array representing the hard edged hexagonal 
+        aperture. 
+
+        Returns
+        -------
+        aperture : Array
+            The aperture represented as a binary float array of 0. and
+            1. representing no transmission and transmission 
+            respectively.
+        """
+        x_centre, y_centre = self.get_centre()
+        number_of_pixels = self.get_npix()
+        maximum_radius = self.get_rmax()
+
+        x, y = get_pixel_positions(number_of_pixels, -x_centre,
+            -y_centre)
+
+        x *= 2 / number_of_pixels
+        y *= 2 / number_of_pixels
+
+        rectangle = (np.abs(x) <= maximum_radius / 2.) \
+            & (np.abs(y) <= (maximum_radius * np.sqrt(3) / 2.))
+
+        left_triangle = (x <= - maximum_radius / 2.) \
+            & (x >= - maximum_radius) \
+            & (np.abs(y) <= (x + maximum_radius) * np.sqrt(3))
+
+        right_triangle = (x >= maximum_radius / 2.) \
+            & (x <= maximum_radius) \
+            & (np.abs(y) <= (maximum_radius - x) * np.sqrt(3))
+
+        hexagon = rectangle | left_triangle | right_triangle
+        return np.asarray(hexagon).astype(float)
+
+
+class PolygonalAperture(Aperture):
+    """
+    """
+    x : float
+    y : float
+    alpha : float
+
+
+    def __init__(self : Layer, pixels : int, pixel_scale : float, 
+            vertices : Matrix, theta : float = 0., phi : float = 0., 
+            magnification : float = 1.) -> Layer:
+        """
+        Parameters
+        ----------
+        pixels : int
+            The number of pixels that the entire compound aperture
+            is to be generated over. 
+        pixel_scale : float
+    
+        vertices : float
+    
+        theta : float, radians
+            The angle that the aperture is rotated from the positive 
+            x-axis. By default the horizontal sides are parallel to 
+            the x-axis.
+        phi : float, radians
+            The angle that the y-axis is rotated away from the 
+            vertical. This results in a sheer. 
+        magnification : float
+            A factor by which to enlarge or shrink the aperture. 
+            Should only be very small amounts in typical use cases.
+        """
+        x, y, alpha = self._vertices(vertices)
+        x_offset, y_offset = self._offset(vertices)
+        self.x = np.asarray(x).astype(float)
+        self.y = np.asarray(y).astype(float)
+        self.alpha = np.asarray(alpha).astype(float)
+        super().__init__(pixels, x_offset, y_offset, theta, phi,
+            magnification, pixel_scale)
+
+
+    def _wrap(self : Layer, array : Vector, order : Vector) -> tuple:
```

Comment:
What does this function do? A little confused here

Reply:
It resets the angles so that they start at the first vertex and then appends the first vertex at the end. It is a useful helper function for the generation.